src/esbuild_plugins/build_routes.ts

```diff
@@ -29,14 +27,22 @@ export const build_routes = (
 				};
 
 				const { html, css: _css, head } = module.default.render();
+
+				// remove any duplicate module imports (in cases where a page uses an island more than once)
+				const deduped_head = Array.from(
+					new Set(
+						head.match(/<script type="module">[\s\S]*?<\/script>/g),
+					),
+				).join("\n");
```

Comment:
This is a great addition!

Are the modules not loaded via the `src` attribute, though?